app/src/main/res/xml/preferences_master.xml

```diff
@@ -71,6 +71,50 @@
                 android:title="@string/file_encryption_password" />
         </PreferenceCategory>
 
+
+        <PreferenceCategory android:title="@string/searchCategory">
+            <ListPreference
+                android:defaultValue="0"
+                android:dialogTitle="@string/max_search_depth"
+                android:entries="@array/pref_arrdisp__search_depth"
+                android:entryValues="@array/pref_arrkeys__search_depth"
+                android:icon="@drawable/ic_baseline_account_tree_24"
+                android:key="@string/pref_key__max_search_depth"
+                android:summary="%1$s"
+                android:title="@string/max_search_depth" />
+            <CheckBoxPreference
+                android:defaultValue="true"
+                android:icon="@drawable/ic_baseline_speaker_notes_off_24"
+                android:key="@string/pref_key__is_show_search_result_on_cancel_enabled"
+                android:summary="@string/show_search_result_on_cancel_summary"
+                android:title="@string/show_search_result_on_cancel" />
+            <android.support.v7.preference.EditTextPreference
+                android:defaultValue=""
+                android:dialogMessage="@string/ignored_search_dir_names_dialog"
+                android:icon="@drawable/ic_baseline_rule_folder_24"
+                android:key="@string/pref_key__ignored_search_dir_names"
+                android:title="@string/ignored_search_dir_names" />
+            <android.support.v7.preference.EditTextPreference
+                android:defaultValue="@string/empty_string"
+                android:dialogMessage="@string/ignored_search_file_names_dialog"
+                android:icon="@drawable/ic_baseline_rule_folder_24"
+                android:key="@string/pref_key__ignored_search_file_names"
+                android:title="@string/ignored_search_file_names" />
+            <android.support.v7.preference.EditTextPreference
+                android:defaultValue="@string/empty_string"
+                android:dialogMessage="@string/search_by_context_have_to_include_dialog"
```

Comment:
Instead of having this and the previous setting, I think it may be better to just have a filename filter directly in the dialog. 

Default filter file would be `*.*`, and a user would be able to enter, say, `*.md`.

Reply:
Yes, it is more convenient